components/scream/src/physics/p3/atmosphere_microphysics.hpp

```diff
@@ -79,13 +79,21 @@ class P3Microphysics : public AtmosphereProcess
         // Exner
         const Spack opmid(pmid(icol,ipack));
```

Comment:
Ok, this is not strictly speaking related to this PR, but while we're at it: we should create refs to views' entries, to save a copy:
```
const auto& opmid = pmid(icol,ipack);
```

Reply:
in the set_variables function below should we also be passing by reference?

Reply:
No. Rule of thumb: builtin types should always be passed by value. Everything that amounts to more than one int/Real/char/pointer is usually better suited to be passed by const reference.

Note: I'm only referring to input values (that the fcn should not modify). If you expect the fcn to modify the argument, then always use reference or pointer, regardless of whether it is a builtin type or not.